### PR TITLE
[TS] added types for pf4 mapper

### LINKED ISSUE #523: [TS] PF4

### PR DESCRIPTION
**Pull request by Hyperkid123**
closes #523

### CODE REVIEW

File: packages/pf4-component-mapper/src/files/wizard.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  description?: ReactNode;
+  isCompactNav?: boolean;
+  inModal?: boolean;
+  setFullWidth?: boolean;
+  setFullHeight?: boolean;
+  isDynamic?: boolean;
+  showTitles?: boolean;
+  crossroads?: string[];
+  fields: Field[];
+  navSchema?: NavSchemaStep[];
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/files/wizard.d.ts:31]: navSchema is passed by internal common wizard

File: packages/pf4-component-mapper/src/files/wizard.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  title?: ReactNode;
+  description?: ReactNode;
+  isCompactNav?: boolean;
+  inModal?: boolean;
+  setFullWidth?: boolean;
+  setFullHeight?: boolean;
+  isDynamic?: boolean;
+  showTitles?: boolean;
+  crossroads?: string[];
+  fields: Field[];
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/files/wizard.d.ts:23]: It isn't a field, it shoul be something like WizardStep

```ts
WizardStep  {
   name: String;
   nextStep: String | Object | Function;
   substepOf: String;
   fields: Field[];
   ... (please see https://data-driven-forms.org/component-example/wizard?mapper=pf4 # Docs for steps )
}
```

File: packages/pf4-component-mapper/src/files/field-array.d.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { ReactNode } from "react";
+import { Field } from "@data-driven-forms/react-form-renderer";
+
+export interface FieldArrayProps {
+  label?: ReactNode;
+  description?: ReactNode;
+  fields: Field[];
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/files/field-array.d.ts:7]: Oh.. does field requires `name`? FieldArray field if there is only one, does not require the name.

File: packages/pf4-component-mapper/src/files/wizard.d.ts (hunk 2 above)
Comment (Hyperkid123) [reply packages/pf4-component-mapper/src/files/wizard.d.ts]: yeah name is required always in `Field` type

File: packages/pf4-component-mapper/src/files/field-array.d.ts (hunk 3 above)
Comment (Hyperkid123) [reply packages/pf4-component-mapper/src/files/field-array.d.ts:7]: yeah name is required always in `Field` type

Comment (rvsia) [reply packages/pf4-component-mapper/src/files/field-array.d.ts:7]: So then it should not be field :D

Comment (Hyperkid123) [reply packages/pf4-component-mapper/src/files/field-array.d.ts]: @rvsia I've added special type for the field array field

**Pull request merged**